examples/paillier_logistic_regression.py

```diff
@@ -0,0 +1,237 @@
+"""
+In this example Alice train a spam classifier on some email dataset she owns.
+She wants to apply it to Bob's personal emails, without
+
+1) asking Bob to send his emails anywhere
+1) leaking information about the learned model or the dataset she has used
+2) letting Bob know which of his e-mails are spam or not.
+
+Alice trains a spam classifier with logistic regression on some data she
+posseses. After learning, generate public and privacy key with a Paillier
+schema. The model is encrypted with the private key. The public key and the
+encrypted models are sent to Bob. Bob applies the encrypted model to his own
+data, obtaining encrypted scores for each email. Bob sends them to Alice.
+Alice decrypts them with the public key and computes the error.
+
+Example inspired by @iamtrask blog post:
+https://iamtrask.github.io/2017/06/05/homomorphic-surveillance/
+
+Dependencies: numpy, sklearn
+"""
+
+import time
+import os.path
+from zipfile import ZipFile
+from urllib.request import urlopen
+from contextlib import contextmanager
+
+import numpy as np
+from sklearn.linear_model import LogisticRegression
+from sklearn.feature_extraction.text import CountVectorizer
+
+import phe as paillier
+
+np.random.seed(42)
+
+# Enron spam dataset hosted by https://cloudstor.aarnet.edu.au
+url = [
+    'https://cloudstor.aarnet.edu.au/plus/index.php/s/RpHZ57z2E3BTiSQ/download',
+    'https://cloudstor.aarnet.edu.au/plus/index.php/s/QVD4Xk5Cz3UVYLp/download'
+       ]
+
+
+def download_data():
+    """Download two sets of Enron1 spam/ham emails if they are not here
+    We will use the first as trainset and the second as testset.
+    Return the path prefix to us to load the data from disk."""
+
+    # The script is meant to work in the main project directory and in the
+    # example directory
+    path_prefix = ''
+    if os.path.isdir('examples'):  # main directory
+        path_prefix = 'examples/'
+    elif os.path.isdir('../examples'):
+        pass
+    else:
+        raise Exception('You are in the wrong path. Move in examples/')
```

Comment:
I think we can clean this download folder piece up further. Calling this script should work from anywhere - just downloading the data into the current directory would be fine.